Integrating ContentProtection Firebolt with AAMP

ALSAMEEMA opened the pull request.

aahmed878 committed:
RDK-56194: Integrating ContentProtectionPlugin with AAMP[DO-NOT MERGE]

Reason for change: Test Patch
Test Procedure: Refer Ticket
Risks: Low
Priority: P1

Signed-off-by: Alsameema <alsameema_ahmedansar@comcast.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=213) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=213) before we can accept your contribution.<br/>**17** out of **21** committers have signed the CLA.<br/><br/>:white_check_mark: anjali-syna<br/>:white_check_mark: Vinish100<br/>:white_check_mark: jfagunde<br/>:white_check_mark: nu641001<br/>:white_check_mark: rajat698<br/>:white_check_mark: Gnanesha<br/>:white_check_mark: psiva01<br/>:white_check_mark: narenr94<br/>:white_check_mark: pstroffolino<br/>:white_check_mark: varatharajan568<br/>:white_check_mark: cpc005<br/>:white_check_mark: p-bond<br/>:white_check_mark: shripadbpersonal<br/>:white_check_mark: aahmed878<br/>:white_check_mark: molakalapalliharipriya<br/>:white_check_mark: srikanthreddybijjam-comcast<br/>:white_check_mark: varshnie<br/>:x: rkandh015<br/>:x: Rajat<br/>:x: rekhap2kandhavelan<br/>:x: dp0000<br/><hr/>**Rajat** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=213) it.</sub>

aahmed878 committed:
RDK-56194: Integrating ContentProtectionPlugin with AAMP[DO-NOT MERGE]

Reason for change: Test Patch
Test Procedure: Refer Ticket
Risks: Low
Priority: P1

Signed-off-by: aahmed878 <Alsameema_Ahmedansar@comcast.com>

ALSAMEEMA committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56194_REBASE2

balasaraswathy-n commented on middleware/externals/playersecmanager/IFirebolt/ContentProtectionFirebolt.cpp:
```diff
@@ -0,0 +1,628 @@
+	}
+}
+
+void ContentProtectionFirebolt::UnSubscribeEvents()
```
Kindly add relevant comments for the routines that are not evaluated

balasaraswathy-n commented on middleware/externals/playersecmanager/IFirebolt/ContentProtectionFirebolt.cpp:
```diff
@@ -0,0 +1,628 @@
+		return false;
+	}
+	auto errorConnect = Firebolt::IFireboltAampAccessor::Instance().Connect(callback);
+	if (!errorConnect) //TODO: Does it always returns positive value? if its an int, does ! check requ?
```
Kindly evaluate

balasaraswathy-n reviewed (commented):
Kindly address review comments

ALSAMEEMA committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56194_REBASE2

aahmed878 committed:
NAMING CHANGES

Signed-off-by: Alsameema <alsameema_ahmedansar@comcast.com>

aahmed878 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56194_REBASE2

Signed-off-by: Alsameema <alsameema_ahmedansar@comcast.com>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e6a4946a5fe7380b173a40320168e778
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"keyId" is passed-by-value as parameter to "std::vector<unsigned char, std::allocator<unsigned char> >::vector(std::vector<unsigned char, std::allocator<unsigned char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""keyId"")" instead of "keyId".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1316](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1316)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
bfe9408053532a53f774a0fe37094778
PRESENT
-->

# Coverity Issue - Uninitialized scalar variable
Using uninitialized value "statusInfo.accessAttributeStatus" when calling "setAccessStatusValue".

_High Impact, CWE-457_
_UNINIT_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1285](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1285)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
11ea3ebf641893bc0825b7841e1d62f4
PRESENT
-->

# Coverity Issue - Double unlock
"~lock_guard" unlocks "thisLock._M_device" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h:99](undefined//blob/undefined/middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h#L99)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
bdf76fe594e27e1f31507865ae9ff27f
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"keyIdArray" is copied in call to copy assignment for class "std::vector<unsigned char, std::allocator<unsigned char> >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""keyIdArray"")" instead of "keyIdArray".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:697](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L697)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
5501f08eb8a6fce118899dfe1cc84ddf
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:890](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L890)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6734ea7c488e180f6967b75fca37759c
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:54](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L54)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
93e84ecebb6d3cc407b05739a838c523
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_downloadConfig>::shared_ptr(std::shared_ptr<_downloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:600](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L600)

aahmed878 committed:
REBASE FAILURES

aahmed878 committed:
REBASE FAILURES

Signed-off-by: Alsameema <alsameema_ahmedansar@comcast.com>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
11f49e49df44bf5b9b57434e014fa506
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"header" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""header"")" instead of "header".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:8781](undefined//blob/undefined/fragmentcollector_mpd.cpp#L8781)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e72d3422538786b1a1d843c2fb5b48b8
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "mpStreamAbstractionAAMP" to a new value. Now the two threads have an inconsistent view of "mpStreamAbstractionAAMP" and updates to fields of "mpStreamAbstractionAAMP" or fields correlated with "mpStreamAbstractionAAMP" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "mpStreamAbstractionAAMP" and the read used to decide whether to modify "mpStreamAbstractionAAMP" with the same set of locks.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:4759](undefined//blob/undefined/priv_aamp.cpp#L4759)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
de88e7411351b948a6e7c6a4f8de09b8
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseResponse" is passed-by-value as parameter to "std::shared_ptr<DrmData>::shared_ptr(std::shared_ptr<DrmData> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseResponse"")" instead of "licenseResponse".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:358](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L358)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
07c1a9d2ec3b7a79e40021c15d3c1381
PRESENT
-->

# Coverity Issue - Double unlock
"~lock_guard" unlocks "thisLock._M_device" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h:106](undefined//blob/undefined/middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h#L106)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1fa85b18d5cf9efd5458b879846fa694
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:461](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L461)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b1e795ed66ce3bfc421e796835b0e465
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:141](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L141)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
c74e360bb1a7ee82d568f8796ffd558f
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"watermarkSessionUpdateCallback" is passed-by-value as parameter to "std::function<void (unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)>::function(std::function<void (unsigned int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)> const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""watermarkSessionUpdateCallback"")" instead of "watermarkSessionUpdateCallback".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:60](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L60)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9d9473ed978e3cbfcc896af108ad5e85
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseServerProxy" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseServerProxy"")" instead of "licenseServerProxy".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:349](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L349)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
fb08fd55657cfebae773ef63ea9c824c
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"data" is copied in call to copy assignment for class "std::vector<std::vector<unsigned char, std::allocator<unsigned char> >, std::allocator<std::vector<unsigned char, std::allocator<unsigned char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""data"")" instead of "data".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:625](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L625)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
819449774a4ad8ae48fe83fadd254f16
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "pMediaStreamContext->numberOfFragmentChunksCached" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.numberOfFragmentChunksCached" is written to with "MediaTrack.mutex" held 3 out of 3 times (2 of these accesses strongly imply that it is necessary).

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:10092](undefined//blob/undefined/fragmentcollector_mpd.cpp#L10092)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d22aa7a4176c62922fd8fb7d33018feb
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:358](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L358)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
fc9a4b25302b6328b7deb2e76672485a
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:64](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L64)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
a5420d7dd3588581ea3031a19fddf89a
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:489](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L489)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
05e4396568d3c880a08c0668721cc41d
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->mAdDuration" without holding lock "PrivateInstanceAAMP.mAdEventQMtx". Elsewhere, "PrivateInstanceAAMP.mAdDuration" is written to with "PrivateInstanceAAMP.mAdEventQMtx" held 1 out of 1 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2283](undefined//blob/undefined/priv_aamp.cpp#L2283)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9c09282eb3222272ef52d416b244d136
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "respData" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1029](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1029)

aahmed878 committed:
CXX VERSION CHANGES

Signed-off-by: aahmed878 <Alsameema_Ahmedansar@comcast.com>

ALSAMEEMA committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56194_REBASE2

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
07c1a9d2ec3b7a79e40021c15d3c1381
PRESENT
-->

# Coverity Issue - Double unlock
"~lock_guard" unlocks "thisLock._M_device" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h:106](undefined//blob/undefined/middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h#L106)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
05e4396568d3c880a08c0668721cc41d
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->mAdDuration" without holding lock "PrivateInstanceAAMP.mAdEventQMtx". Elsewhere, "PrivateInstanceAAMP.mAdDuration" is written to with "PrivateInstanceAAMP.mAdEventQMtx" held 1 out of 1 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2283](undefined//blob/undefined/priv_aamp.cpp#L2283)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1fa85b18d5cf9efd5458b879846fa694
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:461](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L461)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
a5420d7dd3588581ea3031a19fddf89a
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:489](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L489)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e72d3422538786b1a1d843c2fb5b48b8
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "mpStreamAbstractionAAMP" to a new value. Now the two threads have an inconsistent view of "mpStreamAbstractionAAMP" and updates to fields of "mpStreamAbstractionAAMP" or fields correlated with "mpStreamAbstractionAAMP" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "mpStreamAbstractionAAMP" and the read used to decide whether to modify "mpStreamAbstractionAAMP" with the same set of locks.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:4759](undefined//blob/undefined/priv_aamp.cpp#L4759)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9c09282eb3222272ef52d416b244d136
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "respData" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1029](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1029)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
fc9a4b25302b6328b7deb2e76672485a
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:64](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L64)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b1e795ed66ce3bfc421e796835b0e465
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:141](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L141)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
11f49e49df44bf5b9b57434e014fa506
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"header" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""header"")" instead of "header".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:8781](undefined//blob/undefined/fragmentcollector_mpd.cpp#L8781)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
a5420d7dd3588581ea3031a19fddf89a
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:489](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L489)

aahmed878 committed:
L1 TESTS

Signed-off-by: Alsameema <alsameema_ahmedansar@comcast.com>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b1e795ed66ce3bfc421e796835b0e465
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:141](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L141)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
07c1a9d2ec3b7a79e40021c15d3c1381
PRESENT
-->

# Coverity Issue - Double unlock
"~lock_guard" unlocks "thisLock._M_device" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h:106](undefined//blob/undefined/middleware/externals/contentsecuritymanager/ContentSecurityManagerSession.h#L106)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
05e4396568d3c880a08c0668721cc41d
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->mAdDuration" without holding lock "PrivateInstanceAAMP.mAdEventQMtx". Elsewhere, "PrivateInstanceAAMP.mAdDuration" is written to with "PrivateInstanceAAMP.mAdEventQMtx" held 1 out of 1 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2283](undefined//blob/undefined/priv_aamp.cpp#L2283)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9d9473ed978e3cbfcc896af108ad5e85
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseServerProxy" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseServerProxy"")" instead of "licenseServerProxy".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:349](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L349)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
819449774a4ad8ae48fe83fadd254f16
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "pMediaStreamContext->numberOfFragmentChunksCached" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.numberOfFragmentChunksCached" is written to with "MediaTrack.mutex" held 3 out of 3 times (2 of these accesses strongly imply that it is necessary).

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[fragmentcollector_mpd.cpp:10102](undefined//blob/undefined/fragmentcollector_mpd.cpp#L10102)

ALSAMEEMA commented on middleware/externals/CMakeLists.txt:
```diff
@@ -108,11 +120,15 @@ set_target_properties(playerfbinterface PROPERTIES COMPILE_FLAGS "${LIB_EXT_DEFI
+target_link_libraries(playerfbinterface playerjsonobject)
 target_link_libraries(playerfbinterface playerlogmanager)
 target_link_libraries(playerfbinterface baseconversion)
+if(CMAKE_USE_SECCLIENT OR CMAKE_USE_SECMANAGER)
```
Will be addressed as part of bb creation for middleware, as currently firebolt internally requires secmanager, hence firebolt is required wherever is secmanager is included, so, for now, instead of creating a new flag and adding it in multiple bb's, with the use of both SECCLIENT AND SECMANAGER, compiling firebolt code, once these macros are good to be removed, FIREBOLT macro can be added

ALSAMEEMA commented on middleware/externals/playersecmanager/IFirebolt/ContentProtectionFirebolt.cpp:
```diff
@@ -0,0 +1,628 @@
+{
+
+HandleWatermarkEvent(status.sessionId, status.status, status.appId);
+//		mEventHandler = std::make_unique<WatermarkEventHandler>(this);
```
Updated
